test/parallel/test-buffer-read.js

```diff
@@ -0,0 +1,59 @@
+'use strict';
+require('../common');
+const assert = require('assert');
+
+// testing basic buffer read functions
+const buf = Buffer.from([0xa4, 0xfd, 0x48, 0xea, 0xcf, 0xff, 0xd9, 0x01, 0xde]);
+
+function read(buff, funx, args, expected) {
+
+  assert.strictEqual(buff[funx](...args), expected);
+  assert.throws(
+    () => buff[funx](-1),
+    /^RangeError: Index out of range$/
+  );
+
+  assert.doesNotThrow(
+  () => assert.strictEqual(buff[funx](...args, true), expected),
+  'noAssert does not change return value'
```

Comment:
Nit: This would probably be better as something like: `noAssert should not change return value for valid ranges`